### PR TITLE
[hw,i2c,dv]  Add sequence to disable host mode

### PR DESCRIPTION
**Pull request by crteja**
To address requirements in https://github.com/lowRISC/opentitan/issues/17710, 
- Added a new sequence to disable host mode

The following FSM state transitions are observed
`PopFmtFifo` --> `ClockStop`
`HoldStop`  --> `Idle`

Closes: #17710

### CODE REVIEW

File: hw/dv/sv/i2c_agent/i2c_agent_cfg.sv
Comment (marnovandermaas) [inline hw/dv/sv/i2c_agent/i2c_agent_cfg.sv:87]:
```suggestion
    `uvm_field_int(scl_restart,                               UVM_DEFAULT)
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:13]:
```suggestion
  uint num_bytes = 1;
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:30]:
```suggestion
    expected_intr[HostTimeout] = 1;
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:43]:
```suggestion
    fmt_item.start = 1;
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:55]: Nit: same as above

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:62]:
```suggestion
    cfg.m_i2c_agent_cfg.agent_rst = 1;
```

**Review by marnovandermaas — COMMENTED** [review]
Some initial comments, will take a more detailed look later.

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:13]: Updated as per comments

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv]: Updated as per comments

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv]: Updated as per comments

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv]: Updated as per comments

File: hw/dv/sv/i2c_agent/i2c_agent_cfg.sv
Comment (crteja) [reply hw/dv/sv/i2c_agent/i2c_agent_cfg.sv]: Updated as per comments

**Review by marnovandermaas — CHANGES_REQUESTED** [review]
I am seeing failures for this test:
```
UVM_INFO @ 0 ps: reporter [RNTST] Running test i2c_base_test...
UVM_INFO @   1450398 ps: (i2c_base_vseq.sv:254) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.i2c_host_mode_toggle_vseq] 
  initialization is done, DUT/AGENT = Host/Target
UVM_INFO @   2070398 ps: (i2c_base_vseq.sv:525) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.i2c_host_mode_toggle_vseq] 
Programming address in host mode, format flags 0x0110 
  | nakok | rcont |  read |  stop | start |     addr | r/w |
  |     0 |     0 |     0 |     0 |     1 | 00000008 |   W |
UVM_INFO @   2210398 ps: (i2c_base_vseq.sv:525) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.i2c_host_mode_toggle_vseq] 
Programming data in host mode, format flags 0x0601 
  | nakok | rcont |  read |  stop | start |    fbyte |
  |     0 |     0 |     1 |     1 |     0 | 00000001 |
UVM_INFO @   4010398 ps: (i2c_host_mode_toggle_vseq.sv:59) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.i2c_host_mode_toggle_vseq] Disabling host mode
UVM_INFO @   4010398 ps: (i2c_host_mode_toggle_vseq.sv:61) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.i2c_host_mode_toggle_vseq] Resetting agent
UVM_FATAL @   4140398 ps: (i2c_driver.sv:141) uvm_test_top.env.m_i2c_agent.driver [uvm_test_top.env.m_i2c_agent.driver] 
  host_driver, received invalid request: WrData
UVM_INFO @   4140398 ps: (uvm_report_catcher.svh:705) [UVM/REPORT/CATCHER] 
```

I was running this test with xcelium. Any idea why this is?

**Comment by crteja** [comment]
Updated commit to disable transaction comparison from sequence. Dropped https://github.com/lowRISC/opentitan/pull/17942/commits/8bba22ac81940daed157799fa3919ea5b1179129 as it is not required with the update `host_mode_toggle` sequence

**Review by marnovandermaas — APPROVED** [review]
Thanks @crteja, errors seem to be resolved and I am seeing the two transitions you mention being hit. Let's wait for CI to pass and potentially someone else's eyes before merging.

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:5]:
```suggestion
// In DUT=host_mode, create a read/write transaction and randomly disable host_mode mid-transaction by writing ctrl.enablehost = 1'b0.
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv:7]:
```suggestion
// To check the DUT has recovered, start a host_smoke_vseq to check transactions are processed as expected
```

File: hw/ip/i2c/data/i2c_testplan.hjson
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/data/i2c_testplan.hjson:257]:
```suggestion
              - Check if DUT goes to Idle state after Host mode is disabled
              - Check that transactions process normally after recovery by running a smoketest vseq.
```

**Review by hcallahan-lowrisc — APPROVED** [review]
Changes LGTM, just a few nits for comments. Thanks @crteja !

Happy that the transitions are covered based on @marnovandermaas's review comments.

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv]: Updated as per comments in [7c103bf](https://github.com/lowRISC/opentitan/compare/f09ab26e646e0e4375e8e7b2228d322f32ae252b..7c103bfd9522f53485c1b5385fc3f22e57bdcb2c)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv]: Updated as per comments in [7c103bf](https://github.com/lowRISC/opentitan/compare/f09ab26e646e0e4375e8e7b2228d322f32ae252b..7c103bfd9522f53485c1b5385fc3f22e57bdcb2c)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv]: Updated as per comments in [7c103bf](https://github.com/lowRISC/opentitan/compare/f09ab26e646e0e4375e8e7b2228d322f32ae252b..7c103bfd9522f53485c1b5385fc3f22e57bdcb2c)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_mode_toggle_vseq.sv]: Updated as per comments in [7c103bf](https://github.com/lowRISC/opentitan/compare/f09ab26e646e0e4375e8e7b2228d322f32ae252b..7c103bfd9522f53485c1b5385fc3f22e57bdcb2c)

File: hw/ip/i2c/data/i2c_testplan.hjson
Comment (crteja) [reply hw/ip/i2c/data/i2c_testplan.hjson:257]: Updated as per comments in [555310d](https://github.com/lowRISC/opentitan/compare/7c103bfd9522f53485c1b5385fc3f22e57bdcb2c..555310dc38f168b19cc1776a793e504c44b52d5b)

**Comment by crteja** [comment]
CI failures unrelated to the changes

**Comment by marnovandermaas** [comment]
> CI failures unrelated to the changes

Rerunning the failed CI runs because I also think they are unrelated.

**Pull request merged**
